Kernel/FileSystem/File.h

```diff
@@ -142,8 +142,9 @@ class File
     {
         if (Processor::current().in_irq()) {
             // If called from an IRQ handler we need to delay evaluation
-            // and unblocking of waiting threads
-            Processor::deferred_call_queue([this]() {
+            // and unblocking of waiting threads. We need to make sure that
+            // this instance stays alive until then!
+            Processor::deferred_call_queue([this, self = NonnullRefPtr<File>(*this)]() {
```

Comment:
Would this be even better with a `WeakPtr` to the `File`?
If all other references to the file have gone away, nobody should care about the block conditions anyway.